python/pyarrow/tests/test_convert_pandas.py

```diff
@@ -615,6 +615,36 @@ def test_int_object_nulls(self):
         _check_pandas_roundtrip(df, expected=expected,
                                 expected_schema=schema)
 
+    def test_int_object_nulls_option(self):
```

Comment:
It doesn't look like you're using `self` here. Can you make this into a test function and [`pytest.mark.parametrize`](https://docs.pytest.org/en/latest/parametrize.html) it on the `int_dtypes` parameter?

Reply:
We should probably convert this whole module to pytest-style in a separate patch

Reply:
Sure! However, it seems like none of the test methods use `self` and the test classes are just for organizational purposes, so moving it to a test function would be a deviation?

Reply:
Sure, but if we are going to do it eventually then we shouldn't knowingly add to the debt in the name of consistency.

Reply:
I would vote against the pytest-style of a forest of functions. In my experience the lack of organization produces difficult to maintain test modules. Organizing the test methods into several classes helped me figure out which features were tested and how.

An alternative would be to split the tests into several modules (or perhaps several modules inside a subpackage).

Reply:
Grouping by modules (which contain functions) is the solution to that particular problem.